### PR TITLE
mri, initial offsets

### PR DESCRIPTION
**Pull request by osheroff**
allow the consumer to choose an initial offset position on mri.  this can be either `:start`, `:end`, or a fixnum position.  would do the same for the jruby client if it was thought valuable, but I'd need to go poke into the interface.

### CODE REVIEW

File: lib/hermann/consumer.rb [hunk 1]
```diff
@@ -24,6 +24,10 @@ class Consumer
     #
     def initialize(topic, opts = {})
       @topic = topic
+
+      offset = opts.delete(:offset)
+      raise "Bad offset: #{offset}" unless valid_offset?(offset)
```
Comment (rtyler) [inline lib/hermann/consumer.rb]: Would you mind typing this exception and defining it in `lib/hermann/errors.rb`?

File: lib/hermann/consumer.rb [hunk 2]
```diff
@@ -24,6 +24,10 @@ class Consumer
     #
     def initialize(topic, opts = {})
```
Comment (rtyler) [inline lib/hermann/consumer.rb]: Will you update the RDoc to include the :`offset` option documentation including mention of `:start` and `:end`

**Comment by rtyler** [comment]
After this change is merged, I'll cut a new release to rubygems.org since my magic auto-deploy stuff isn't entirely working correctly

**Comment by osheroff** [comment]
updated, thanks for the review!

File: spec/consumer_spec.rb [hunk 3]
```diff
@@ -36,6 +37,11 @@
         let(:topic) { '' }
         it_behaves_like 'an error condition'
       end
+
+      context 'with a bad offset' do
+        let(:offset) { :foo }
+        it_behaves_like 'an error condition'
```
Comment (rtyler) [inline spec/consumer_spec.rb]: Looks like this test needs to be updated per Travis failures

**Comment by osheroff** [comment]
yup, sorry.  fixed now.

**Comment by rtyler** [comment]
0.23.242 contains this change

**Comment by osheroff** [comment]
awesome, thanks much @rtyler

**Pull request merged**